Sources/Kommunicate/Classes/KMConversationService.swift

```diff
@@ -82,22 +82,24 @@ public class KMConversationService: KMConservationServiceable,Localizable {
                 if present {
                     let groupID = Int(truncating: channel?.key ?? 0)
                     let response = Response(success: true, clientChannelKey: clientId, error: nil)
-                    guard let currentAssignee = self.assigneeUserIdFor(groupId: groupID),
-                          let newAssignee = conversation.conversationAssignee,
-                          !newAssignee.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty,
-                          newAssignee != currentAssignee else {
-                        completion(response)
-                        return
-                    }
-                    self.assignConversation(groupId: groupID, to: newAssignee) { result in
-                        switch result {
-                        case .success:
-                            completion(response)
-                        case .failure(let error):
-                            let errorResponse = Response(success: false, clientChannelKey: clientId, error: error)
-                            completion(errorResponse)
+                    
+                    if let currentAssignee = self.assigneeUserIdFor(groupId: groupID), let newAssignee = conversation.conversationAssignee {
+                        if (!(newAssignee.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty) && newAssignee != currentAssignee) {
+                            self.assignConversation(groupId: groupID, to: newAssignee) { result in
+                                switch result {
+                                case .success:
+                                    completion(response)
+                                case .failure(let error):
+                                    let errorResponse = Response(success: false, clientChannelKey: clientId, error: error)
+                                    completion(errorResponse)
+                                }
+                            }
                         }
                     }
+                    self.updateGroupMetadata(groupId: NSNumber(value: groupID), channelKey: "", metadata: self.groupMetadata) { response in
```

Comment:
Try updating assignee and metadata together. See if it's working. Both the calls will be going in parallel.

Reply:
Have added the changes.

Reply:
We also need `dispatchGroup.wait()` and then do something on completion, right?